package/kernel/mac80211/patches/rt2x00/620-01-rt2x00-fix-clock-init-for-rt2800-SoC.patch

```diff
@@ -0,0 +1,91 @@
+From a56662b417852a8895ad5851c4a91dc6a7a0fa8d Mon Sep 17 00:00:00 2001
+From: Shiji Yang <yangshiji66@outlook.com>
+Date: Sun, 22 Dec 2024 17:06:59 +0800
+Subject: [PATCH 1/3] rt2x00: fix clock init for rt2800 SoC
+
+It seems that the rt2800 SoC built-in WMAC requires correct system
+bus clock to work properly. However, the clk_get_sys() no longer
+work with the newly introduced rt2800 clock driver. Hence rt2x00
+driver always uses a dafault value 125 MHz to initialize US_CYC_CNT
+register. This patch allows us to specify the "bus" and "xtal"
+clock source from the device tree.
+
+Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
+---
+ .../net/wireless/ralink/rt2x00/rt2800lib.c    | 19 ++-----------------
+ drivers/net/wireless/ralink/rt2x00/rt2x00.h   |  3 ++-
+ .../net/wireless/ralink/rt2x00/rt2x00soc.c    | 16 +++++++++++++---
+ 3 files changed, 17 insertions(+), 21 deletions(-)
+
+--- a/drivers/net/wireless/ralink/rt2x00/rt2800lib.c
++++ b/drivers/net/wireless/ralink/rt2x00/rt2800lib.c
+@@ -2439,7 +2439,7 @@ static void rt2800_config_lna_gain(struc
+ 
+ static inline bool rt2800_clk_is_20mhz(struct rt2x00_dev *rt2x00dev)
+ {
+-	return clk_get_rate(rt2x00dev->clk) == 20000000;
++	return clk_get_rate(rt2x00dev->xtal_clk) == 20000000;
+ }
+ 
+ #define FREQ_OFFSET_BOUND	0x5f
+@@ -6271,22 +6271,7 @@ static int rt2800_init_registers(struct
+ 		rt2x00_set_field32(&reg, US_CYC_CNT_CLOCK_CYCLE, 125);
+ 		rt2800_register_write(rt2x00dev, US_CYC_CNT, reg);
+ 	} else if (rt2x00_is_soc(rt2x00dev)) {
+-		struct clk *clk = clk_get_sys("bus", NULL);
+-		int rate;
+-
+-		if (IS_ERR(clk)) {
+-			clk = clk_get_sys("cpu", NULL);
+-
+-			if (IS_ERR(clk)) {
+-				rate = 125;
```

Comment:
@paraka Hi! Would you mind have a review on this patch? Is it a correct fix? I noticed my MT7620 board always use rate 125 to initialize the US_CYC_CNT.

Reply:
@DragonBluep : You should just get wmac clock here. That is the reason of that introduced in the driver as periph clock with xtal as parent. I don't understand why are you trying to get the bus and xtal clocks here... If the clock plan is correct (it should be) only node index for wmac in dts file must be used. No other change should be needed in dts. If there is something to fix regarding the clock it should be done at the driver side and just consume correct clock here.